### PR TITLE
Iss 686

### PR DESCRIPTION
**Pull request by burkmarr**
This is ready for pulling unless you find any issues @johnvanbreda.
Pull request for issue https://github.com/BiologicalRecordsCentre/iRecord/issues/686

### CODE REVIEW

File: prebuilt_forms/quick_species_maps.php [hunk 1]
```diff
@@ -179,9 +179,16 @@ public static function get_form($args, $nid, $response = NULL) {
...
+
+    $r .= '<div style="margin:0.5em 0 0.5em 0"><input type="checkbox" checked id="showInstructions" style="margin-right:1em">'. lang::get('Show instructions');
+    $r .= '<input type="checkbox" id="showLegend" style="margin:0 1em 0 3em" checked>'. lang::get('Show legend') . '</div>';
     $r .= '<p id="instruct">' . lang::get('Click on the + buttons in the grid to add species layers to the map. You can add up to {1} layers at a time.',
-        count($args['indicia_species_layer_slds']));
-    $r .= '<p id="instruct2" style="display: none">' . lang::get('Use the - buttons to permanently remove layers, or untick the box in the legend to temporarily hide them.');
+         count($args['indicia_species_layer_slds']));
+    $r .= lang::get('Use the - buttons to permanently remove layers, or untick the box in the legend to temporarily hide them.');
+
+    // $r .= '<p id="instruct">' . lang::get('Click on the + buttons in the grid to add species layers to the map. You can add up to {1} layers at a time.',
```
Comment (johnvanbreda) [inline prebuilt_forms/quick_species_maps.php]: Any need to leave the commented lines in?

File: prebuilt_forms/js/quick_species_maps.js [hunk 2]
```diff
@@ -86,18 +81,40 @@ $(document).ready(function () {
...
       indiciaData.mapdiv.map.addLayer(layer);
       layers.push({layer:layer,key:key});
-      if (layers.length>indiciaData.indiciaSpeciesLayer.slds.length) {
+      //if (layers.length>indiciaData.indiciaSpeciesLayer.slds.length) {
+      if (layers.length>10) {
         remove=layers.splice(0,1);
         img = $('tr#row'+remove[0].key).find('img');
-        img.attr('src', img.attr('src').replace('delete.png','add.png'));
-        img.removeClass('on-map');
+        // Image may not be found, e.g. if filtes have been applied.
```
Comment (johnvanbreda) [inline prebuilt_forms/js/quick_species_maps.js]: Typo (being very fussy!)

File: prebuilt_forms/js/quick_species_maps.js [hunk 3]
```diff
@@ -86,18 +81,40 @@ $(document).ready(function () {
           {isBaseLayer: false, sphericalMercator: true, singleTile: true, opacity: 0.5});
       indiciaData.mapdiv.map.addLayer(layer);
       layers.push({layer:layer,key:key});
-      if (layers.length>indiciaData.indiciaSpeciesLayer.slds.length) {
+      //if (layers.length>indiciaData.indiciaSpeciesLayer.slds.length) {
```
Comment (johnvanbreda) [inline prebuilt_forms/js/quick_species_maps.js]: Is this supposed to be 10? Why 10?

File: prebuilt_forms/js/quick_species_maps.js [hunk 4]
```diff
@@ -49,13 +49,8 @@ $(document).ready(function () {
...
       return;
     }
-    // Toggle through instructions to get the user started
-    if (sequence===0) {
-      $('#instruct').slideUp();
-      $('#instruct2').show();
-    } else if (sequence===1) {
-      $('#instruct2').slideUp();
-    }
+
```
Comment (johnvanbreda) [inline prebuilt_forms/js/quick_species_maps.js:74]: No double line spaces

File: prebuilt_forms/quick_species_maps.php (hunk 1 above)
Comment (burkmarr) [reply prebuilt_forms/quick_species_maps.php]: No sorry - untidy of me. I will remove.

File: prebuilt_forms/js/quick_species_maps.js (hunk 2 above)
Comment (burkmarr) [reply prebuilt_forms/js/quick_species_maps.js]: Will sort.

File: prebuilt_forms/js/quick_species_maps.js (hunk 4 above)
Comment (burkmarr) [reply prebuilt_forms/js/quick_species_maps.js:74]: Is this okay John?

**Comment by burkmarr** [comment]
Hi @johnvanbreda. Can you have another look. After looking at the code again, I realised that I had already sorted most of those - I made a couple of commits after pull request, so things got a little confused.

**Pull request merged**